tests/sync/util.js

```diff
@@ -50,21 +50,21 @@ const syncFromDisk = require('../../sync/syncFromDisk');
 
 /**
```

Comment:
All of the type definitions seem to appear in both `tests/sync/util.js` and `sync/course-db.js`. Is there any way to have just one copy of these somewhere? I think they might already be out of sync (`Question.dependencies` is only in `sync/course-db.js`?).

Reply:
Yeah, I think we can consolidate most of them in `sync/course-db.js`!

Reply:
Ahhh, I remember why I had them separate - the test utils types have many properties marked as optional so that you don't have to include as many of them in type data, whereas the types for the syncing code itself do *not* have as many optional properties because the defaults applied when loading JSON will generally the properties are defined.

If this were actually TypeScript, we could use the full type system and define the syncing types in terms of the other types but with particular properties as optional, buuuut it's not actually TypeScript, so we can't 🙁.

Reply:
Haha, no worries.